Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -958,6 +958,11 @@ protected IEnumerable<TItem> DisplayData
         /// </summary>
         [Parameter] public Action<TItem> NewItemDefaultSetter { get; set; }
 
+        /// <summary>
+        /// Function that, if set, is called to create new instance of an item. If left null a default constructor will be used.
+        /// </summary>
+        [Parameter] public Func<TItem> NewItemFactory { get; set; }

```

Comment:
Maybe name it `NewItemCreator`? A _factory_ kinda points to a similar design pattern but it doesn't feel like it in this case.

Reply:
@stsrki Maybe... Factory normally creates more than one type so you may be right. It does not matter to me in the end as long as I can specify it.

Reply:
Also I I am not sure if we should pass a parameter with the method. It is possible to pass up to 16 params but that would require a LOT of overloads so a single parameter which holds a class with other parameters would be the preferred approach.

Reply:
Changed to creator in new commit.